### PR TITLE
feat: Ulo download button #61

### LINKED ISSUE #61: ULO.py: Make a download button for the selected data.
### Bug Description

The underlying data from the figures need to be downloaded.

### To Reproduce

4. See error.

### Expected Behaviour

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by yokat**
## Description

Please include a summary of the change(s).
You should also include motivation (why did this change required?) and which issue this pull request fixes (what problem does it solve?).

Don't forget to reference all other pull requests which are associated to this one below: e.g.

closes #61 
relates road86/repository_name#issue_number

## Testing (delete if not applicable)

Please describe the new tests that have been added for this change along with any additional instructions required to reproduce a successful test and/or a list of any relevant details for your test configuration.

## Checklist

- [ ] I have read the road86 Contribution Guide.
- [ ] I have checked all commit message styles match the requested structure.
- [ ] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have assigned at least one reviewer.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] My changes generate no new warnings.
- [ ] I have made corresponding changes to the documentation.
- [ ] New database changes have been committed.

### CODE REVIEW

**Comment by mixmixmix** [comment]
I've checked and it seem to be working! Would it be easy to customise the name of the file? E.g. it could have the selected upazila name and today's date?

**Review by mixmixmix — CHANGES_REQUESTED** [review]
Could have the selected upazila name and today's date be added to the output file?

**Comment by yokat** [comment]
I added this feature. Can I decide, that it is done, or do you need to have a second look?

**Comment by ChasNelson1990** [comment]
> I added this feature. Can I decide, that it is done, or do you need to have a second look?

You will also need a new review if you push new commits :-)

File: pages/ULO.py [hunk 1]
```diff
@@ -405,11 +412,39 @@ def selectULO(SelDiv, SelDis, SelUpa, SelDiseases, sdate, edate):
...
-    return SelDiv, SelDis, SelUpa, vDis, vUpa, dislis, Diseases, minSelDate, maxSelDate, startDate, endDate, disabSelDate, figULORep, figULOSick, figULODead
+    return SelDiv, SelDis, SelUpa, vDis, vUpa, dislis, Diseases, minSelDate, maxSelDate, startDate, endDate, disabSelDate, figULORep, figULOSick, figULODead, exportbutton
+    
+
+@callback(   
+    Output("ULO_export", "data"),
+    Input ('btn_csv', 'n_clicks'),
+    State ('ULO_Upazila','value'),
+    prevent_initial_call=True,
+)
```
Comment (mixmixmix) [inline pages/ULO.py:440]: I get an error
```
Traceback (most recent call last):
  File "/Users/mix/repos/bahis-dash/pages/ULO.py", line 446, in export
    return dcc.send_data_frame(bahis_subdata.to_csv, str(SelUpa) + "_export_" + str(date.today()) + ".csv")
AttributeError: 'list' object has no attribute 'to_csv'
```

**Review by mixmixmix — CHANGES_REQUESTED** [review]

File: pages/ULO.py (hunk 1 above)
Comment (yokat) [reply pages/ULO.py:439]: :(

**Review by mixmixmix — CHANGES_REQUESTED** [review]
It works now, however I wonder how difficult would that be for the data and the filename to have actual names in place of geocodes?

**Comment by yokat** [comment]
Would work, but I would keep the geocode, because some upazila names are not unique.

**Comment by mixmixmix** [comment]
> Would work, but I would keep the geocode, because some upazila names are not unique.

a simple fix would be just to add a number to the end of upazila name. What do you think? A lot of utility value for a small fix!

**Comment by yokat** [comment]
like this?

**Review by mixmixmix — APPROVED** [review]
brill

**Pull request merged**